students/MicahBraun/Lesson 04/mailroom_2.py

```diff
@@ -0,0 +1,287 @@
+# ------------------------------------------------------------------------
+# NAME: MICAH BRAUN
+# PROJECT: mailroom.py
+# PURPOSE: Working with arrays(nested lists), manipulating data
+# DATE: 06/06/2018
+#
+# DESCRIPTION: Program is menu-driven user interface for interacting
+#              with a nested list of stored information. User is able
+#              to add names to a list (not currently saved upon program
+#              exit), append values to fields, check against current
+#              fields, create mock "Thank You" e-mails based on user-
+#              -input and associated fields, and generate tabular display
+#              of data for viewing.
+# ------------------------------------------------------------------------
+import time  # time to supply today's date for letters
+import os  # for default working directory
+import locale  # for currency formatting
+
+# Data -------------------------------------------------------------------
+locale.setlocale(locale.LC_ALL, '')
+today = time.strftime("%m/%d/%Y")
+
+spacing = '- ' * 56  # formatting for DONOR header
+donor_dict = {'Columns': ['NAMES', 'DONATION AMOUNT', 'NUMBER OF GIFTS', 'AVG. GIFTS'],
+              'Rudolph Soares': ['Rudolph Soares', 1335, 3, 445],  # Donor info
+              'Josef Mistretta': ['Josef Mistretta', 7500, 4, 1875],
+              'Joye Agostini': ['Joye Agostini', 600, 2, 300],
+              'Rachelle Levan': ['Rachelle Levan', 12100, 5, 2420],
+              'Vena Ussery': ['Vena Ussery', 4000, 8, 500],
+              'Efrain Lager': ['Efrain Lager', 795, 9, 88.34],
+              'Mee Heine': ['Mee Heine', 4600, 4, 1150],
+              'Tanya Essex': ['Tanya Essex', 75000, 2, 37500]}
```

Comment:
I would highly recommend storing individual donations rather than calculated values and doing your calculations where you generate the report, perhaps building another data structure there to format the data for the report. This will simplify a lot of your other code